dockstore-client/src/main/java/io/dockstore/client/cli/nested/WesLauncher.java

```diff
@@ -0,0 +1,203 @@
+     * @param workflowEntry The entry path, (i.e. github.com/myRepo/myWorkflow:version)
+     * @param workflowParamPath The path to a file to be used as an input JSON. (i.e. /path/to/file.json)
+     * @param filePaths A list of paths to files to be attached to the request.
+     */
+    public static void launchWesCommand(WorkflowClient workflowClient, String workflowEntry, String workflowParamPath, List<String> filePaths) {
+
+        // Get the workflow object associated with the provided entry path
+        final Workflow workflow = getWorkflowForEntry(workflowClient, workflowEntry);
+
+        // Can take the following values:
+        // 1. A TRS URL returning the raw primary descriptor file contents
+        // 2. TODO: A path to a file in the 'attachments' list
+        String workflowUrl = combineTrsUrlComponents(workflowClient, workflowEntry, workflow);
+
+        // A JSON object containing a key/value pair that points to the test parameter file in the 'attachments' list
+        // The key is WES server implementation specific. e.g. {"workflowInput":"params.json"}.
+        File workflowParams = fetchFile(workflowParamPath).orElse(null);
+
+        // A list of supplementary files that are required to run the workflow. This may include any/all of the following:
+        // 1. The primary descriptor file
+        // 2. Secondary descriptor files
+        // 3. Test parameter files
+        // 4. Any other files referenced by the workflow
+        // TODO: Allow users to specify a directory to upload?
+        // TODO: Automatically attach all files referenced in remote Dockstore entry?
+        List<File> workflowAttachment = fetchFiles(filePaths);
+
+        // The descriptor type
+        String workflowType = workflow.getDescriptorType().getValue();
+
+        // The workflow version
+        String workflowTypeVersion = createWorkflowTypeVersion(workflowType);
+
+        try {
+            RunId response = workflowClient
+                .getWorkflowExecutionServiceApi()
+                .runWorkflow(
+                    workflowParams,
+                    workflowType,
+                    workflowTypeVersion,
+                    TAGS,                       // TODO: User specified tags?
+                    WORKFLOW_ENGINE_PARAMETERS, // TODO: User specified engine parameters?
+                    workflowUrl,
+                    workflowAttachment);
+
+            String runID = response.getRunId();
+            out("Launched WES run with id: " + runID);
+            wesCommandSuggestions(runID);
+        } catch (io.openapi.wes.client.ApiException e) {
+            LOG.error("Error launching WES run", e);
+        }
+    }
+
+    /**
+     * Obtains the workflow on Dockstore
+     *
+     * @param workflowClient The WorkflowClient
+     * @param workflowEntry The workflow entry (i.e. github.com/myRepo/myWorkflow:version)
+     * @return A Workflow object for the corresponding workflow+version
```

Comment:
What happens if the entry doesn't exist? Throws an exception or returns null? If null, please update this; if an exception, I'd add it to line 91 (even though we don't always document runtime exceptions).

Reply:
The `WorkflowClient` throws an exception and the program exits if the entry doesn't exist.

Reply:
It looks like an uncaught exception, which is why the program exits. This can be fine, but we generally catch these (see line 86).

Reply:
Oops, I miswrote. An we raise an error for the user, rather than throw an exception. This occurs here: https://github.com/dockstore/dockstore-cli/blob/8167b87d138806083bad92f5cffede2a8bafb2f2/dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java#L320

So, in this case, there is no exception just an intentional `System.exit()` and message.